src/Shared/StaticWebAssets/ManifestStaticWebAssetFileProvider.cs

```diff
@@ -320,6 +323,8 @@ internal sealed class StaticWebAssetManifest
 
         public StaticWebAssetNode Root { get; set; } = null!;
 
+        [UnconditionalSuppressMessage("Trimming", "IL2026", Justification = "Public properties preserved with DynamicDependency")]
+        [DynamicDependency(JsonSerializerSettings, typeof(StaticWebAssetManifest))]
```

Comment:
You will need to make this recursive on the inner types:

- StaticWebAssetMatch
- StaticWebAssetPattern